### PR TITLE
Atlas sync

### PR DESCRIPTION
**Pull request by lenucksi**
This GitHub Actions pipeline synchronizes our repo content on changes to master to the O'Reilly Atlas repository and merges in the local specifics to enable rendering the content properly.

As a last step a build on O'Reilly is triggered using the Gem described here: http://docs.atlas.oreilly.com/api.html

Secrets live in the project and run off of my account first. Having a bot account for that purpose probably would do no harm though.

Once this works, extending it to the other repos is a quick fix.

**TODO:** I might need to add runs using the actual `GITHUB_SHA` of the current state to enable using this as a pull request check. Or might need to add a second pipeline running against pull request checks separating the sync run on pushes to master and the pull request check run.

### CODE REVIEW

File: .github/workflows/SyncToAtlas.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@master 
+    - name: Add O'Reilly remotes
+      run: |
+        git remote add oreilly git@git.atlas.oreilly.com:lenucksi/learningpath-introduction.git
+    - uses: fregante/setup-git-token@v1
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (rrrutledge) [inline .github/workflows/SyncToAtlas.yml]: Where are these secrets stored?

File: .github/workflows/SyncToAtlas.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    - name: enable SSH-Key
+      env: 
+        OREILLYKEY: ${{ secrets.OREILLYKEY }}
+      run: |
+        mkdir -p $HOME/.ssh
+        #ls -al ~/.ssh
+        echo -e "$OREILLYKEY" > ~/.ssh/id_rsa
+        chmod 600 ~/.ssh/id_rsa
+        eval `ssh-agent -s`
```
Comment (rrrutledge) [inline .github/workflows/SyncToAtlas.yml]: Just curious - why is `eval` needed here instead of just running `ssh-agent -s` directly?

File: .github/workflows/SyncToAtlas.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        ssh-add ~/.ssh/id_rsa
+        ssh-keyscan git.atlas.oreilly.com >> ~/.ssh/known_hosts
+        #    - name: checkout master
+        #      run: git checkout master
```
Comment (rrrutledge) [inline .github/workflows/SyncToAtlas.yml]: These commented-out lines no longer needed?

File: .github/workflows/SyncToAtlas.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+jobs:
+  atlas:
+
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@master 
+    - name: Add O'Reilly remotes
+      run: |
+        git remote add oreilly git@git.atlas.oreilly.com:lenucksi/learningpath-introduction.git
```
Comment (rrrutledge) [inline .github/workflows/SyncToAtlas.yml]: Is this some fork of the actual learning path?

File: .github/workflows/SyncToAtlas.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        # Get our master into a different branch. Merge our master-branch into the O'Reilly master.
+        # Push our new merged master into the O'Reilly master.
+        # Party!
+        git fetch oreilly
+        git checkout -b oreilly-master oreilly/master
+        git merge -m "Merge GitHub content to Atlas" -v --commit --log master 
+        git push oreilly oreilly-master:master
+    - name: Trigger build on Atlas
+      run: |
+        gem install atlas
```
Comment (rrrutledge) [inline .github/workflows/SyncToAtlas.yml]: This is [Hashicorp Atlas](https://rubygems.org/gems/atlas/)?  Is that related to O'Reilly Atlas?

Comment (StingRayZA) [reply .github/workflows/SyncToAtlas.yml]: I think this should be `gem install atlas-api`...
http://docs.atlas.oreilly.com/api.html#atlas-gem

File: .github/workflows/SyncToAtlas.yml (hunk 1 above)
Comment (StingRayZA) [reply .github/workflows/SyncToAtlas.yml]: @rrrutledge they're stored in the project itself. this particular one is a builtin - but the others can be found by navigating to Settings | Secrets on the project...

File: .github/workflows/SyncToAtlas.yml (hunk 2 above)
Comment (StingRayZA) [reply .github/workflows/SyncToAtlas.yml]: `ssh-agent` outputs some env variables when it runs. if you wrap it in an `eval`, it activates those and makes the agent available to subsequent calls.
https://linux.die.net/man/1/ssh-agent (look under _There are two main ways to get an agent set up_)

File: .github/workflows/SyncToAtlas.yml (hunk 1 above)
Comment (lenucksi) [reply .github/workflows/SyncToAtlas.yml]: Exactly what @StingRayZA says.

File: .github/workflows/SyncToAtlas.yml (hunk 3 above)
Comment (lenucksi) [reply .github/workflows/SyncToAtlas.yml]: Yes, indeed.

File: .github/workflows/SyncToAtlas.yml (hunk 1 above)
Comment (rrrutledge) [reply .github/workflows/SyncToAtlas.yml]: Thanks.

File: .github/workflows/SyncToAtlas.yml (hunk 2 above)
Comment (rrrutledge) [reply .github/workflows/SyncToAtlas.yml]: Got it.

File: .github/workflows/SyncToAtlas.yml (hunk 3 above)
Comment (rrrutledge) [reply .github/workflows/SyncToAtlas.yml]: Thanks.

File: .github/workflows/SyncToAtlas.yml (hunk 4 above)
Comment (rrrutledge) [reply .github/workflows/SyncToAtlas.yml]: Thanks.  Will anyone else ever need to see what is in _Atlas_?  It looks like I, myself, can't view this repo, for example.

File: .github/workflows/SyncToAtlas.yml (hunk 5 above)
Comment (rrrutledge) [reply .github/workflows/SyncToAtlas.yml]: Yes!  https://rubygems.org/gems/atlas-api/

File: .github/workflows/SyncToAtlas.yml (hunk 4 above)
Comment (lenucksi) [reply .github/workflows/SyncToAtlas.yml]: > Thanks. Will anyone else ever need to see what is in _Atlas_? 

Aside from people at O'Reilly? No. Some of it is likely O'Reilly IP anyway.

> It looks like I, myself, can't view this repo, for example.

Now you should be able to.

File: .github/workflows/SyncToAtlas.yml (hunk 5 above)
Comment (lenucksi) [reply .github/workflows/SyncToAtlas.yml]: Thanks for catching that! Looks like a case of incomplete commandline copypasta.
```suggestion
        gem install atlas-api
```

File: .github/workflows/SyncToAtlas.yml (hunk 4 above)
Comment (lenucksi) [reply .github/workflows/SyncToAtlas.yml]: > Thanks. Will anyone else ever need to see what is in _Atlas_? 

Aside from people at O'Reilly? No. Some of it is likely O'Reilly IP anyway.

> It looks like I, myself, can't view this repo, for example.

Now you should be able to.
